lib/order.rb

```diff
@@ -0,0 +1,98 @@
+class Order
+  attr_accessor :products, :customer, :fulfillment_status
+  attr_reader :id
+
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id
+    @products = products
+    @customer = customer
+    @fulfillment_status = fulfillment_status
+    # fulfillment_status = :pending <-- this is the syntax to set a default value for a positional argument
+
+    valid_fulfillment_statuses = [:pending, :paid, :processing, :shipped, :complete]
+    if !valid_fulfillment_statuses.include?(fulfillment_status)
+      raise ArgumentError, "Fulfillment status is not valid."
+    end
+    # A fulfillment_status, a symbol, one of :pending, :paid, :processing, :shipped, or :complete
+    # If no fulfillment_status is provided, it will default to :pending
+    # If a status is given that is not one of the above, an ArgumentError should be raised
+
+  end
+
+  def total
+    # calculate the total cost of the products incl 7.5% tax; round to two decimal places
+    order_total = (products.values.sum * 1.075).truncate(2)
+    return order_total
+  end
+
+  def add_product(product_name, product_price)
+    # add a new product to the product collection; if a product with the same name has already been added to the products, an ArgumentError should be raised
+    if products.keys.include?(product_name)
+      raise ArgumentError, "This product has already been ordered."
+    end
+
+    products[product_name] = product_price
+  end
+
+  def remove_product(product_name)
+    # WRITING THIS METHOD WAS OPTIONAL
+    # remove a specified product from the collection; if no product with that name was found, an ArgumentError should be raised
+    if products.keys.include?(product_name)
+      products.delete(product_name)
+    else
+      raise ArgumentError, "Product not found."
+    end
+  end
+
+  def self.products_helper(products_from_csv_row) # <-- convert string to hash
```

Comment:
Nice decision to pull this logic into a method! Also, good call on making it a class method!